src/browzine-summon-adapter.js

```diff
@@ -98,56 +124,103 @@ angular.module("summonApp.directives")
-    template = template.replace(/{{browzineWebLink}}/g, browzineWebLink);
-    template = template.replace(/{{bookIcon}}/g, bookIcon);
+    template = template.replace(/{wording}/g, wording);
+    template = template.replace(/{browzineWebLink}/g, browzineWebLink);
+    template = template.replace(/{browzineWebLinkText}/g, browzineWebLinkText);
+    template = template.replace(/{bookIcon}/g, bookIcon);
 
     return template;
   };
 
-  return {
-    link: function(scope, element, attributes) {
-      if(!shouldEnhance(scope)) {
-        return;
-      }
+  function getScope(documentSummary) {
+    return angular.element(documentSummary).scope();
+  };
 
-      var endpoint = getEndpoint(scope);
+  function resultsWithBrowZine(documentSummary) {
+    var scope = getScope(documentSummary);
 
-      http.get(sce.trustAsResourceUrl(endpoint)).then(function(response) {
-        var data = getData(response);
-        var browzineWebLink = getBrowZineWebLink(data);
-        var coverImageUrl = getCoverImageUrl(data, response);
+    if(!shouldEnhance(scope)) {
+      return;
+    }
 
-        if(browzineWebLink) {
-          var template = buildTemplate(data, browzineWebLink, bookIcon);
-          element.find(".docFooter .row:first").append(template);
-        }
+    var endpoint = getEndpoint(scope);
 
-        if(coverImageUrl) {
-          element.find(".coverImage img").attr("src", coverImageUrl);
-        }
-      });
-    }
+    $.getJSON(endpoint, function(response) {
+      var data = getData(response);
+      var browzineWebLink = getBrowZineWebLink(data);
+      var coverImageUrl = getCoverImageUrl(data, response);
+
+      if(browzineWebLink) {
+        var template = buildTemplate(data, browzineWebLink);
+        $(documentSummary).find(".docFooter .row:eq(0)").append(template);
+      }
+
+      if(coverImageUrl) {
+        $(documentSummary).find(".coverImage img").attr("src", coverImageUrl);
+      }
+    });
+  };
+
+  return {
+    resultsWithBrowZine: resultsWithBrowZine,
+    getScope: getScope,
+    shouldEnhance: shouldEnhance,
+    getEndpoint: getEndpoint,
+    getData: getData,
+    getBrowZineWebLink: getBrowZineWebLink,
+    getCoverImageUrl: getCoverImageUrl,
+    buildTemplate: buildTemplate,
+  };
+}());
+
+$(function() {
+  if(typeof browzine === "undefined" || browzine === null) {
+    return;
+  }
+
+  var results = document.querySelector("#results");
+  var config = {
+    attributes: true,
+    childList: true,
+    characterData: true,
+    subtree: true,
   };
-}]);
 
-// End BrowZine-Summon Integration Code
+  //Enhance any documentSummary elements present before the observer starts
+  var documentSummaries = results.querySelectorAll(".documentSummary");
+
+  Array.prototype.forEach.call(documentSummaries, function(documentSummary) {
+    browzine.search.resultsWithBrowZine(documentSummary);
+  });
+
+  var observer = new MutationObserver(function(mutations) {
```

Comment:
Neat. I'd not come across MutationObserver before. TIL

Reply:
Yep, it saved our bacon, hehe. Thankfully, we get IE11+ with it too.